README.md

```diff
@@ -262,77 +201,220 @@ builder.toValue();
   '{"op":"and","content":[{"op":"in","content":{"fieldName":"name","value":"Tim"}},{"op":"gt","content":{"fieldName":"age","value":"19"}}]}',
 );
 ```
 
-This will throw an error if the provided value is not a valid SQON or JSON string.
+This differs from the default function because it will accept any input. The use case for this function is when you have a JS object that could potentially be a valid SQON, you can pass this `uncheckedObject` to `SQONBuilder.from(uncheckedObject)` and validation will be performed before the `SQONBuilder` is returned, if possible.
+
+If the provided string cannot be parsed from JSON a `SyntaxError` will be thrown.
+
+If the provided object or parsed string is not a valid SQON, a [`ZodError`](https://zod.dev/ERROR_HANDLING) will be thrown.
+
+### SQONBuilder Object
+
+The `SQONBuilder` type is an object that stores a `SQON` object and exposes functions to modify that `SQON`, returning a new `SQONBuilder` object. This allows a user to chain function calls together to create complex filters one step at a time.
+
+All filter and combination methods listed above can be called from the returned `SQONBuilder` object:
+* In: `.in(fieldName: string, value: ArrayFilterValue) => SQONBuilder`
+* GreaterThan: `.gt(fieldName: string, value: number) => SQONBuilder`
+* LesserThan: `.lt(fieldName: string, value: number) => SQONBuilder`
+* And: `.and(sqon: SQONBuilder | SQON) => SQONBuilder`
+* Or: `.in(sqon: SQONBuilder | SQON) => SQONBuilder`
+* Not: `.in(sqon: SQONBuilder | SQON) => SQONBuilder`
+
+Example: `const mySqon: SQON = SQONBuilder.in('name', ['Jim']).or(SQONBuilder.gt('age', 25).lt('score', 50)).toValue();`
+
+All builder methods are side-effect free: modifications made with to a `SQONBuilder` are additive on top of a cloned SQON from the previous builder, so that the original builder is not modified. This can let you, for example, create a builder as a base SQON to make variations on in your application:
+
+```ts
+const base = SQONBuilder.in('name', ['Jim']);
+const ageRestricted = base.gt('age', 25);
+// {"op":"and","content":[{"op":"in","content":{"fieldName":"name","value":["Jim"]}},{"op":"gt","content":{"fieldName":"age","value":25}}]}
+
+const scoreRestricted = base.lt('score', 50);
+// {"op":"and","content":[{"op":"in","content":{"fieldName":"name","value":["Jim"]}},{"op":"lt","content":{"fieldName":"score","value":50}}]}
+```
+
+As filters and combinations are added to a `SQONBuilder`, the `SQON` maintained will be [reduced](#reduce-sqon), if possible. This will result in matching filters and combinations to be combined, and empty combinations to be removed.
+
+In addition to the filter and combination functions, there are some additional methods that can modify the filters stored in the existing SQON, and then two methods for outputing the stored `SQON` as either a string with [`.toString()`](#tostring) or a value with [`.toValue()`](#tovalue)
+
+#### ToString
+`builder.toString() => string`
```

Comment:
```suggestion
`builder.toString()` returns a `string`
```

Reply:
Need better clarity on this document of outlining the member methods of the default export vs the members on a SQONBuilder object...

In the examples I provided, the `SQONBuilder` token specifically refers to the default export, but most of its methods return an object of __type__ `SQONBuilder`, hence the confusion. For `toString` and `toValue`, that function is not a member of the default exporter: ie. `SQONBuilder.tostring()` is not a function. However, a previously created builder has these methods: `SQONBuilder(mySqon).toString()`.

This is all to say I can't accept this change as is, but some overhauling of the whole readme is needed to clarify this.

Reply:
I see! that makes sense now.